flow360/component/simulation/framework/base_model.py

```diff
@@ -532,7 +531,7 @@ def _convert_dimensions_to_solver(
         extra: List[Any] = None,
     ) -> dict:
         solver_values = {}
-        self_dict = deepcopy(self.__dict__)
+        self_dict = self.__dict__
```

Comment:
this change can be in conflict with line 547 (modifying self). I think we can create classmethod for any extra dimensional fields and remove this logic from here.

Reply:
I put these extra field in a separate dict and we loop both of the dicts together.